static/app/utils/useDismissAlert.spec.tsx

```diff
@@ -0,0 +1,116 @@
+import {reactHooks} from 'sentry-test/reactTestingLibrary';
+
+import localStorage from 'sentry/utils/localStorage';
+import useDismissAlert from 'sentry/utils/useDismissAlert';
+
+jest.mock('sentry/utils/localStorage');
+jest.useFakeTimers();
+
+const mockSetItem = localStorage.setItem as jest.MockedFunction<
+  typeof localStorage.setItem
+>;
+const mockGetItem = localStorage.getItem as jest.MockedFunction<
+  typeof localStorage.getItem
+>;
+
+const key = 'test_123';
+const now = new Date('2020-01-01');
+
+describe('useDismissAlert', () => {
+  beforeEach(() => {
+    jest.setSystemTime(now);
+
+    mockSetItem.mockReset();
+    mockGetItem.mockReset();
+  });
+
+  it('should return a stable ref for the dismiss() function', () => {
+    const {result, rerender} = reactHooks.renderHook(useDismissAlert, {
+      initialProps: {key},
+    });
+
+    const initialRef = result.current.dismiss;
+
+    rerender();
+
+    expect(result.current.dismiss).toEqual(initialRef);
+  });
+
+  it('should not be dismissed if there is no value in localstorage', () => {
+    mockGetItem.mockReturnValue(null);
+
+    const hook = reactHooks.renderHook(useDismissAlert, {
+      initialProps: {key},
+    });
+    const {result} = hook;
+
+    expect(result.current.isDismissed).toBeFalsy();
+  });
+
+  it('should be dismissed if there is any value in localstorage and no expiration', () => {
+    mockGetItem.mockReturnValue('some value');
+
+    const {result} = reactHooks.renderHook(useDismissAlert, {
+      initialProps: {key},
+    });
+
+    expect(result.current.isDismissed).toBeTruthy();
+  });
+
+  it('should set the current timestamp into localstorage when an alert is dismissed', () => {
+    const {result} = reactHooks.renderHook(useDismissAlert, {
+      initialProps: {key},
+    });
+
+    reactHooks.act(() => {
+      result.current.dismiss();
+      jest.runAllTicks();
+    });
+
+    expect(mockSetItem).toHaveBeenCalledWith(
+      key,
+      JSON.stringify(now.getTime().toString())
+    );
+  });
+
+  it('should be dismissed if the timestamp in localstorage is older than the expiration', () => {
+    const today = new Date('2020-01-01');
```

Comment:
should we reuse the const `now` here?

Reply:
i used a new variable so it's local to the test and easier to spot the difference between the two dates.